cmd/dymint/commands/start.go

```diff
@@ -130,10 +131,14 @@ func startInProcess(config *cfg.NodeConfig, tmConfig *tmcfg.Config, logger log.L
 }
 
 func checkGenesisHash(config *tmcfg.Config) error {
-	if len(genesisHash) == 0 || config.Genesis == "" {
+	if len(genesisHash) == 0 {
 		return nil
 	}
 
+	if config.Genesis == "" {
```

Comment:
u should check this before `if len(genesisHash) == 0`
u want to return error if genesis path not provided